core/domain/user_services_test.py

```diff
@@ -1363,11 +1429,102 @@ def test_validate_creator_dashboard_display_pref(self):
     def test_guest_has_not_fully_registered(self):
         self.assertFalse(user_services.has_fully_registered_account(None))
 
-    def test_cannot_create_new_user_with_existing_user_id(self):
-        with self.assertRaisesRegexp(
-            Exception, 'User %s already exists.' % self.owner_id
-        ):
-            user_services.create_new_user(self.owner_id, self.OWNER_EMAIL)
+    def test_save_user_auth_for_user_auth_enabled_creates_new_auth_entry(self):
+        with self.swap(feconf, 'ENABLE_USER_AUTH_MODEL', True):
+            new_user_id = user_models.UserSettingsModel.get_new_id('')
+            new_gae_id = 'new_gae_id'
+            new_user_pin = '123'
+            user_auth = user_services.UserAuth(
+                user_id=new_user_id,
+                gae_id=new_gae_id,
+                pin=new_user_pin
+            )
+            user_services._save_user_auth_details(user_auth) # pylint: disable=protected-access
+            fetched_user_auth_model = user_models.UserAuthModel.get_by_id(
+                new_user_id)
+            self.assertEqual(
+                fetched_user_auth_model.gae_id, user_auth.gae_id)
+            self.assertEqual(fetched_user_auth_model.pin, user_auth.pin)
+
+    def test_save_user_auth_user_auth_disabled_does_not_create_new_entry(self):
```

Comment:
test_save_user_auth__with_user_auth_flag_disabled_does_not_create_new_entry, Generally lets try to make the test names a proper sentence
You can maybe even say test_save_user_auth__with_flag_disabled_does_not_create_new_entry